Sources/CvPlayerAI.cpp

```diff
@@ -12852,9 +12852,58 @@ int CvPlayerAI::AI_unitPropertyValue(UnitTypes eUnit, PropertyTypes eProperty) c
 	return iValue;
 }
 
-/********************************************************************************/
-/* 	City Defenders						24.07.2010				Fuyu			*/
-/********************************************************************************/
+int CvPlayerAI::AI_unitBuildingValue(UnitTypes eUnit, CvArea* pArea) const
+{
+	FAssertMsg(eUnit != NO_UNIT, "Unit is not assigned a valid value");
+	FAssertMsg(eUnitAI != NO_UNITAI, "UnitAI is not assigned a valid value");
```

Comment:
There is no eUnitAI, this line can just be removed. It appears it is valid for pArea to be null (this should be noted in a comment on the function, or by giving a default parameter of null in the declaration). Although its kind of weird for this function to just return 0 if pArea is null. When is that useful? Instead it should either evalute the building for all areas if area is null, OR it should just not be called at all, and require area to not be null.